src/query/frontend/semantic/symbol_generator.cpp

```diff
@@ -302,6 +307,17 @@ SymbolGenerator::ReturnType SymbolGenerator::Visit(Identifier &ident) {
   if (scope.in_skip || scope.in_limit) {
     throw SemanticException("Variables are not allowed in {}.", scope.in_skip ? "SKIP" : "LIMIT");
   }
+
+  if (scope.in_exists && (scope.visiting_edge || scope.in_node_atom)) {
+    auto has_symbol = HasSymbol(ident.name_);
```

Comment:
I think there might be a bug here.

I tried the following query in order to break this part:
```
MATCH (n),(m) 
WITH n 
WHERE exists((n)-[]->(m)) 
RETURN n;
```

And it returns `An unknown exception occurred, this is unexpected. The real message should be in the database logs`

The problem might be that you are checking all the scopes before if there is a symbol `m` and that shouldn't be the case. I am not 100% sure how these scopes work with `WITH` clause, but maybe it makes sense to only check recent scope. That seems like reason why it later explodes somewhere, haven't figured out where yet.

Reply:
@Josipmrden I am sorry for giving wrong review here but important thing:
**I think this shouldn't throw and should be allowed.** More details below.

Reply:
I am not sure if that is the reason why you decided to throw error in rule based planner, to make this MVP implementation.

Reply:
Resolved as we won't address it now